main/src/com/google/refine/operations/row/RowKeepMatchedOperation.java

```diff
@@ -0,0 +1,74 @@
+
+package com.google.refine.operations.row;
+
+import static com.google.refine.operations.OperationDescription.row_keep_matching_brief;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import com.google.refine.browsing.Engine;
+import com.google.refine.browsing.EngineConfig;
+import com.google.refine.browsing.FilteredRows;
+import com.google.refine.browsing.RowVisitor;
+import com.google.refine.history.HistoryEntry;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.model.changes.RowRemovalChange;
+import com.google.refine.operations.EngineDependentOperation;
+
+public class RowKeepMatchedOperation extends EngineDependentOperation {
+
+    @JsonCreator
+    public RowKeepMatchedOperation(
+            @JsonProperty("engineConfig") EngineConfig engineConfig) {
+        super(engineConfig);
+    }
+
+    @Override
+    protected String getBriefDescription(Project project) {
+        return row_keep_matching_brief();
+
+    }
+
+    @Override
+    protected HistoryEntry createHistoryEntry(Project project, long historyEntryID) throws Exception {
+        Engine engine = createEngine(project);
+        FilteredRows filteredRows = engine.getAllFilteredRows();
+
+        Set<Integer> matchedRows = new HashSet<>();
+
+        filteredRows.accept(project, new RowVisitor() {
+
+            @Override
+            public void start(Project project) {
+            }
+
+            @Override
+            public void end(Project project) {
+            }
+
+            @Override
+            public boolean visit(Project project, int rowIndex, Row row) {
+                matchedRows.add(rowIndex);
+                return false;
+            }
+        });
+        List<Integer> rowsToRemove = new ArrayList<>(project.rows.size() - matchedRows.size());
+        for (int i = 0; i < project.rows.size(); i++) {
+            if (!matchedRows.contains(i)) {
+                rowsToRemove.add(i);
+            }
+        }
+        return new HistoryEntry(
+                historyEntryID,
+                project,
+                row_keep_matching_brief() + " - " + rowsToRemove.size() + " rows removed",
```

Comment:
This is still a mix of languages: if OpenRefine is running in Italian and a translator added an Italian translation, then this will generate something like:
"Mantenere le righe corrispondenti - 34 rows removed"
This should be avoided, since it contains two different languages in the same phrase.

As I tried to explain before, we need to make the whole message translatable. This requires introducing a new message id, with a parameter containing the number of rows removed. To understand how to add parameters to a message, you could take inspiration from the `recon_clear_similar_cells_desc` message. Could you please do that?

Reply:
For reference the section in our doc regarding [Adding a new string for localization](https://openrefine.org/docs/technical-reference/translating-ui#front-end-development)